### DISCUSSION TITLE
Jetstream & Fortify

### DISCUSSION THREAD

**Discussion by malsgit** [spatie/laravel-multitenancy#450] [General]
Hi 
I'm trying to create a multi tenancy app, I have Jetstream and Fortify to handle login, registration etc.

On getting it working and a tenancy databse setup with standard migrations users, migrations, sessions tables etc.

When I try to register on the tenent site it is reading trying to register on the landlord DB rather than the Tenants DB, 

what steps do i need to take to get the user to register on the tenants DB rather than attemping to register on the landlord DB??

I've googled to no avial, I'm assuming someone has done this at some point...

Any help much appreciated...

**Comment by boguslaw** [comment]
Hi. 

My tests showed that the application connects to the database from the default configuration (.env), not to the landlord database.

A similar problem was described here https://github.com/spatie/laravel-multitenancy/discussions/439 . My temporary solution based on comments from that discussion is:

(for example in AppServiceProvider)
```
use Illuminate\Support\Facades\DB;

... 

    public function boot(): void
    {
        DB::setDefaultConnection(
            config('multitenancy.tenant_database_connection_name')
        );
    }
```

**Comment by m7vm7v** [comment]
My approach is similar. But rather than using the AppServiceProvider I have rewritten the Spatie's Database Task, making it slightly more the Spatie's way.

In your config/multitenancy.php update this -

```
    'switch_tenant_tasks' => [
        \Spatie\Multitenancy\Tasks\PrefixCacheTask::class,
       ...
        \YOUR_NAMESPACE\Support\SwitchTenantTasks\SwitchTenantDatabaseTask::class,
       ...
        \Spatie\Multitenancy\Tasks\SwitchRouteCacheTask::class,
    ],
```

Then create the SwitchTenantDatabaseTask - 
```
<?php

namespace YOUR_NAMESPACE\Support\SwitchTenantTasks;

use Illuminate\Support\Facades\DB;
use Spatie\Multitenancy\Tasks\SwitchTenantDatabaseTask as SpatieSwitchTenantDatabaseTask;

class SwitchTenantDatabaseTask extends SpatieSwitchTenantDatabaseTask
{
    protected function setTenantConnectionDatabaseName(?string $databaseName)
    {
        parent::setTenantConnectionDatabaseName($databaseName);

        $tenantConnectionName = is_null($databaseName)
            ? $this->landlordDatabaseConnectionName()
            : $this->tenantDatabaseConnectionName();

        DB::setDefaultConnection($tenantConnectionName);
    }
}

```

**Reply by m7vm7v** [reply]
I am still under the impression that should happen automatically from the package but not happening for my application. I will be really happy if we could have an explanation if we are accidentally doing something wrong or that is expected behaviour. I can confirm that this happens on the latest Laravel 9 with Jetstream and Fortify.

**Comment by masterix21** [comment]
Any package assumes that its models use the default connection: it happens with Sanctum, Fortify, Spatie Medialibrary... all packages.

Using a package like `spatie/laravel-multitenancy` with many databases means that a developer knows what he is doing because Laravel won't manage your connection switch automatically, never.

For example, how `Team` model in Jetstream should know the right database if isn't the developer instructs it to work?

`spatie/laravel-multitenancy` comes with lots of tasks to reduce the developing time, but doesn't exists a best practice to solve all kinds of problems. The fast reply is always to switch the default connection, but it should be better to handle the proper connection for each model, keeping the landlord connection always as the default connection. IMHO.

**Comment by leobeal** [comment]
old thread, but this might help someone: 
You can simply configure the `connection` in config/session.php to use the tenant connection.
